### ISSUE TITLE
Wrong loop counters

### ISSUE THREAD

**Issue by walisser** [level1wendell/memcpy_sse#4]
I'm sure you'll catch this one, but the copy loop is unrolled 8 times so you need to step the pointers by 1024 bytes and not 128. Should be a nice speedup!

With this change memcpy_sse is almost twice as fast as memcpy (gcc 5.4 on Linux/Nehalem). FWIW, this CPU doesn't exhibit and performance change between x32 and x64.

**Comment by Sakari-N** [comment]
The loop copies 8 x 16 byte blocks each iteration so incrementing source and destination pointers by 128 bytes on each iteration is correct.

**Comment by walisser** [comment]
Of course, I would read 128-bit instructions and think they were 128 BYTES. That puts it back to about 10%/35% slower (x32/x64) than libc memcpy.

**Comment by tpruzina** [comment]
[This](https://github.com/level1wendell/memcpy_sse/blob/master/testmem_modified.c#L13) should use restrict keywords since compiler has to assume that areas might overlap and cant pipeline reads and writes simultaneously. It also assumes that that source/dest are aligned and performs horribly if they are not.

I have played around this a bit, these are the results (64bit gentoo, gcc 9.0.0 20180530, 7700k):

```
gcc cpy.c && ./a.out  | sort -n
     1608783 clocks [SOCPY]
     1994590 clocks [SSE]
     3486523 clocks [INTMAX]
    28298007 clocks [NAIVE]

gcc -march=native cpy.c && ./a.out  | sort -n
     1620174 clocks [SOCPY]
     1905277 clocks [SSE]
     3447400 clocks [INTMAX]
    28032391 clocks [NAIVE]

gcc -O3 cpy.c && ./a.out  | sort -n
      248548 clocks [SOCPY]
      267219 clocks [NAIVE]
      504281 clocks [INTMAX]
      797964 clocks [SSE]

gcc -O3 -march=native cpy.c && ./a.out  | sort -n
      156088 clocks [INTMAX]
      180580 clocks [NAIVE]
      502264 clocks [SOCPY]
      738958 clocks [SSE]
```

[src gist](https://gist.github.com/tpruzina/440a60c1ae910e0c7be850bbb6d4d766)
